acme/acme/challenges.py

```diff
@@ -279,7 +280,13 @@ def validation_domain_name(self, name: str) -> str:
         :rtype: str
 
         """
-        return f"{self.LABEL}.{name}"
+        name = f"{self.LABEL}.{name}"
+        try:
+            name = str(dns.resolver.resolve(name, "CNAME")[0])[:-1]
```

Comment:
Not sure what this gets us, I think this might actually break some things.

In what situation will the `name` not be `_acme-challenge.{domain being validated}`? Even when following cnames, that's really only relevant for the value of the record, not the label.

Reply:
As stated in the description: When you host your domain with an unsupported provider but still want to use the DNS01 challenges.

Reply:
I guess I'm wondering why would certbot need to do that, that seems like the ACME server side things right?

Reply:
The server does that. The client up to now does not. Without the CNAME resolution, the code here https://github.com/certbot/certbot/blob/master/certbot-dns-route53/certbot_dns_route53/_internal/dns_route53.py#L76 and in other DNS plugins looks for a wrong zone and fails.